uuid.go

```diff
@@ -305,6 +312,33 @@ func (u *UUID) Scan(src interface{}) error {
 	return fmt.Errorf("uuid: cannot convert %T to UUID", src)
 }
 
+// Value implements the driver.Valuer interface.
+func (u NullUUID) Value() (driver.Value, error) {
+	if !u.Valid {
+		return nil, nil
+	}
+	// Delegate to UUID Value function
+	return u.UUID.Value()
+}
+
+// Scan implements the sql.Scanner interface.
+func (u *NullUUID) Scan(src interface{}) error {
+	if src == nil {
+		*u = NullUUID{Nil, false}
```

Comment:
Suggestion, consider using keyed struct fields here, it will make the code more readable:

```Go
*u = NullUUID{UUID: Nil, Valid: false}
```